contracts/staking/StakeManager.sol

```diff
@@ -97,8 +99,32 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     _;
   }
 
-  function stake(uint256 amount, address signer, bool isContract) external {
-    stakeFor(msg.sender, amount, signer, isContract);
+  // TopUp heimdall fee
+  function topUpForFee(uint256 validatorId, uint256 heimdallFee) public {
+    require(heimdallFee >= MIN_DEPOSIT_SIZE, "Minimum amount is 1 Matic");
```

Comment:
Might be a better idea to require this in `_topUpForFee` since you are calling that function from `stakeFor` as well.

Reply:
it's okay to not have it in case of `stakeFor` because that function is assumed to be Sybil attack protected